### PR TITLE
Images migrated to azure blob

### LINKED ISSUE #452: Save all images to azure instead of DB
**_Date_**
12.03.21

**_Time_**
15:30

**_Description_**
Change storage for users and events images from DB to Azure Storage

**_Tasks_**
- [x] Rewrite code that working with events photos
- [x] Rewrite code that working with users photos
- [x] Save and download images to remote Azure Storage

### PR DESCRIPTION
**Pull request by dyats**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk

### Second Level Review

- [ ] @sand0

## Summary of issue

Need to change storage for events and users photos from DB to Azure Blob Storage

## Summary of change

- Written interaction with Azure Blob Storage
- Written custom value resolvers for mapper profiles
- Fixed services interaction
- Fixed Unit tests

## Testing approach

- Created events and users and photos was uploaded to Blob Storage
- Updated events and users and photos was overwritten to Blob Storage
- Photos fetching correctly
- Works with remote Blob Storage

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EventsExpress.Core/DTOs/EventDTO.cs [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using Azure.Storage.Blobs;
+using EventsExpress.Core.Extensions;
+using EventsExpress.Core.IServices;
+using EventsExpress.Core.Services;
```
Comment (sand0) [inline EventsExpress.Core/DTOs/EventDTO.cs]: do we really need all these here?

File: EventsExpress.Core/Services/UserService.cs [hunk 2]
```diff
@@ -274,22 +270,16 @@ public async Task ChangeRole(Guid userId, Guid roleId)
...
-            if (user.Photo != null)
-            {
-                await _photoService.Delete(user.Photo.Id);
-            }
-
             try
             {
-                user.Photo = await _photoService.AddPhoto(avatar);
+                await _photoService.AddUserPhoto(avatar, user.Id);
                 Update(user);
```
Comment (sand0) [inline EventsExpress.Core/Services/UserService.cs]: I guess we don't need a call to update here, 'cause we don't change our db here

File: EventsExpress.Test/MapperTests/UserMapperProfileTests.cs [hunk 3]
```diff
@@ -1,24 +1,163 @@
...
+    using EventsExpress.Core.DTOs;
+    using EventsExpress.Core.IServices;
+    using EventsExpress.Db.Entities;
+    using EventsExpress.Db.Enums;
+    using EventsExpress.Mapping;
+    using EventsExpress.Test.MapperTests.BaseMapperTestInitializer;
+    using EventsExpress.ViewModels;
+    using Microsoft.Extensions.DependencyInjection;
+    using Moq;
+    using NUnit.Framework;
```
Comment (sand0) [inline EventsExpress.Test/MapperTests/UserMapperProfileTests.cs]: just move it outside of namespaces

File: EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs [hunk 4]
```diff
@@ -45,21 +44,14 @@ protected override void Initialize()
...
             evnt = new Event
             {
                 Id = validEventId,
                 DateFrom = DateTime.Today,
                 DateTo = DateTime.Today,
                 Description = "...",
-                PhotoId = validPhotoId,
-                Photo = photo,
+                /*PhotoId = validPhotoId,
+                Photo = photo,*/
```
Comment (sand0) [inline EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs]: just remove commented code

File: EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs [hunk 5]
```diff
@@ -99,7 +91,7 @@ protected override void Initialize()
                     EventId = validEventId,
             };
 
-            Context.Photos.Add(photo);
+            // Context.Photos.Add(photo);
```
Comment (sand0) [inline EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs]: and here

File: EventsExpress/Controllers/EventController.cs [hunk 6]
```diff
@@ -4,6 +4,7 @@
 using System.Threading.Tasks;
 using AutoMapper;
 using EventsExpress.Core.DTOs;
+using EventsExpress.Core.Extensions;
```
Comment (sand0) [inline EventsExpress/Controllers/EventController.cs]: do we need this?

**Review by sand0 — CHANGES_REQUESTED** [review]

File: EventsExpress/Controllers/EventController.cs (hunk 6 above)
Comment (dyats) [reply EventsExpress/Controllers/EventController.cs]: Nope. Removed.

File: EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs (hunk 5 above)
Comment (dyats) [reply EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs]: Done.

File: EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs (hunk 4 above)
Comment (dyats) [reply EventsExpress.Test/ServiceTests/EventScheduleServiceTests.cs]: Done.

File: EventsExpress.Test/MapperTests/UserMapperProfileTests.cs (hunk 3 above)
Comment (dyats) [reply EventsExpress.Test/MapperTests/UserMapperProfileTests.cs]: Okay.

File: EventsExpress.Core/Services/UserService.cs (hunk 2 above)
Comment (dyats) [reply EventsExpress.Core/Services/UserService.cs]: Yep. Done.

File: EventsExpress.Core/DTOs/EventDTO.cs (hunk 1 above)
Comment (dyats) [reply EventsExpress.Core/DTOs/EventDTO.cs]: No.

**Review by iuriikhrystiuk — APPROVED** [review]

**Review by sand0 — APPROVED** [review]

**Comment by sand0** [comment]
I've updated app service sr=etings on the azure test to use new connection string for bob storage.

But.. shouldn't we drop all the existing events in DB to avoid fails when fetching photos for them? 
@iuriikhrystiuk @DennYats

**Pull request merged**